### PR TITLE
multiple categories

### PR DESCRIPTION
**Pull request by okkez**
connect to https://github.com/bm-sms/daimon-news-misc/issues/154

### CODE REVIEW

File: app/models/categorisation.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Categorisation < ActiveRecord::Base
```
Comment (tricknotes) [inline app/models/categorisation.rb]: `Categorization` ?

Comment (okkez) [inline app/models/categorisation.rb]: See http://ejje.weblio.jp/content/categorisation

File: app/models/post.rb [hunk 2]
```diff
@@ -23,8 +23,10 @@ def with_ordered
 
   scope :published, -> { where("published_at <= ?", Time.current) }
   scope :order_by_recent, -> { order(published_at: :desc, id: :asc) }
+  scope :categorized, ->(category) { joins(:categories).where("categories.id" => category.id) }
```
Comment (tricknotes) [inline app/models/post.rb]: Prefer `categorized_by`, `categorized_as` or something?

File: app/views/editor/posts/index.html.haml [hunk 3]
```diff
@@ -22,7 +22,10 @@
       %tr{ class: post.published? ? nil : 'post--unpublished' }
         %td= post.public_id
         %td= post.title
-        %td= post.category&.name
+        %td
+          %ul
+            - post.categories.pluck(:name).each do |name|
```
Comment (tricknotes) [inline app/views/editor/posts/index.html.haml]: `AR::Relation#pluck` fires SQL query. 
If there are no reason for using `pluck`, can you use eager loading?

File: app/models/categorisation.rb (hunk 1 above)
Comment (cafedomancer) [inline app/models/categorisation.rb]: Maybe `categorisation` is British spelling and `categorization` is American spelling.

Comment (tricknotes) [inline app/models/categorisation.rb]: That't right.
Are there any reason to use British spelling?

Comment (okkez) [inline app/models/categorisation.rb]: I've forgotten reason.
I will change spelling.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -15,6 +15,6 @@ def show
   private
 
   def current_category
-    @post.category
+    @post.categories.first
```
Comment (tricknotes) [inline app/controllers/posts_controller.rb:18]: Is the `ordered` scope unnecessary?

File: app/views/editor/posts/index.html.haml (hunk 3 above)
Comment (okkez) [inline app/views/editor/posts/index.html.haml]: `post.categories.map(&:name).each ...` ?

File: app/controllers/posts_controller.rb (hunk 4 above)
Comment (tricknotes) [inline app/controllers/posts_controller.rb:18]: Ah, it is already applied 💡 

``` ruby
has_many :categorisations, -> { ordered }, dependent: :destroy
```

File: app/views/layouts/application.html.haml [hunk 5]
```diff
@@ -27,11 +27,11 @@
 
       .wrappable
         %span.xxx
-          = render 'application/breadcrumbs'
+          = render 'application/breadcrumbs', options: {}
```
Comment (tricknotes) [inline app/views/layouts/application.html.haml]: Can `options: {}` be optional?

File: db/migrate/20160719005841_create_categorisations.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      t.references :category, foreign_key: true, null: false
+      t.references :post, foreign_key: true, null: false
+      t.integer :order, null: false
+
+      t.index %i(post_id category_id), unique: true
+
+      t.timestamps null: false
+    end
+
+    Post.all.each do |post|
```
Comment (tricknotes) [inline db/migrate/20160719005841_create_categorisations.rb]: Shouldn't reference model from migration.
see: [マイグレーション作成時のチェックポイント](http://qiita.com/nay3/items/ef773006cd7f815a07cd#2-%E3%82%A2%E3%83%97%E3%83%AA%E3%82%B1%E3%83%BC%E3%82%B7%E3%83%A7%E3%83%B3%E3%81%A7%E5%AE%9A%E7%BE%A9%E3%81%97%E3%81%9F%E3%82%AF%E3%83%A9%E3%82%B9%E3%82%92%E7%9B%B4%E6%8E%A5%E4%BD%BF%E3%82%8F%E3%81%AA%E3%81%84)

File: app/views/layouts/application.html.haml (hunk 5 above)
Comment (okkez) [inline app/views/layouts/application.html.haml]: How to do that? Could you tell me?

File: db/migrate/20160719005841_create_categorisations.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        post.categories << category
+        post.save!
+      end
+    end
+
+    remove_column :posts, :category_id
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
```
Comment (tricknotes) [inline db/migrate/20160719005841_create_categorisations.rb]: We can add down migration using SQL.

File: app/views/layouts/application.html.haml (hunk 5 above)
Comment (tricknotes) [inline app/views/layouts/application.html.haml]: Such as using [`local_assigns`](http://api.rubyonrails.org/classes/ActionView/Template.html#method-i-local_assigns) .

``` haml
-# application/_breadcrumbs.html.haml
- options = local_assigns[:options] || {}
```

File: db/structure.sql [hunk 8]
```diff
@@ -2,16 +2,12 @@
 -- PostgreSQL database dump
 --
 
--- Dumped from database version 9.5.3
```
Comment (tricknotes) [inline db/structure.sql:5]: Do you use PostgreSQL 9.5.3?

File: test/factories/post.rb [hunk 9]
```diff
@@ -7,7 +7,11 @@
 
     trait :whatever do
       site
-      category { create(:category, site: site) }
+      after :create do |post, _evaluator|
+        post.categorisations << build(:categorisation, :whatever, site: post.site)
+        post.save!
+        post.reload
```
Comment (tricknotes) [inline test/factories/post.rb:13]: Are there any reasons to call `reload`?

File: app/views/layouts/application.html.haml (hunk 5 above)
Comment (okkez) [inline app/views/layouts/application.html.haml]: Thanks! :bow:

File: db/structure.sql (hunk 8 above)
Comment (okkez) [inline db/structure.sql:5]:
```
$ psql --version
psql (PostgreSQL) 9.5.3
```

Comment (tricknotes) [inline db/structure.sql:5]: nice!
I guess any PostgreSQL's configuration is different from this `structure.sql`.
I'm happy unnecessary diffs aren't there.

File: app/views/editor/posts/index.html.haml (hunk 3 above)
Comment (tricknotes) [inline app/views/editor/posts/index.html.haml]: How about the following?

``` haml
- post.categories.each do |category|
  = category.name
```

**Comment by tricknotes** [comment]
It seems there is a redundant margin.

**Comment by okkez** [comment]
We will merge #596 includes this PR.

**Pull request closed without merging**